File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 18/08/2026, 14:57:41
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, maxProfit = 0, 0

        for right in range(len(prices)):
            if prices[right] > prices[left]:
                maxProfit = max(maxProfit, prices[right] - prices[left])
            elif prices[right] < prices[left]:
                left = right
        
        return maxProfit

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 18/08/2026, 14:57:33
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int left = 0;

        for (int right = 0; right < prices.length; right++) {
            if (prices[right] < prices[left]) {
                left = right;
            } else {
                maxProfit = Math.max(maxProfit, prices[right] - prices[left]);
            }
        }

        return maxProfit;
    }
}